src/components/AssetHeader/MoreActionsDrawer.tsx

```diff
@@ -52,10 +54,12 @@ export const MoreActionsDrawer: React.FC<MoreActionsDrawerProps> = ({
     onClose()
   }, [assetId, dispatch, onClose])
 
-  const handleToggleSpam = useCallback(() => {
+  const handleToggleSpam = useCallback(async () => {
     dispatch(preferences.actions.toggleSpamMarkedAssetId(assetId))
     onClose()
-  }, [assetId, dispatch, onClose])
+
+    if (!isSpamMarked) await moralisReportSpam(assetId)
```

Comment:
see note in PR description re: this no works upstream but might as well 
https://discord.com/channels/819584798443569182/1402737023114023093/1402737023114023093

Reply:
Think that's fine! Gut feel is that reports should obviously not count towards use (and is borked upstream anyway) - in regular use I could see toggling on and off and on with two lifecycles (e.g something like "yeah I wasn't sure it was a spam but it definitely is") which would definitely be justify-able.
Else could be malicious use but meh, that's a feature not a bug IMO!

Other option is we could track report already submitted by user but feels like a lot of lift for something we're not sure yet will even work

Reply:
With that being said, since not implemented upstream anyway, going to comment this out 

4336e4c8d0